File: backends/java/zappy-adapters/src/main/java/com/zappymart/adapters/graphql/WishlistPayload.java
package com.zappymart.adapters.graphql;

import com.zappymart.domain.catalogue.Product;
import com.zappymart.domain.shared.UserError;

import java.util.List;

public record WishlistPayload(List<Product> products, List<UserError> errors) {
}

File: backends/dotnet/src/Zappy.Adapters.GraphQL/Payloads/WishlistPayload.cs
using Zappy.Application;
using Zappy.Domain;

namespace Zappy.Adapters.GraphQL;

public sealed class WishlistPayload(IReadOnlyList<Product> products, IReadOnlyList<UserError> errors)
{
    public IReadOnlyList<Product> Products { get; } = products;

    public IReadOnlyList<UserError> Errors { get; } = errors;

    public static WishlistPayload From(WishlistResult result) => new(result.Products, result.Errors);
}
